core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java

```diff
@@ -88,16 +91,16 @@ void build(List<SubstrateConfigBuildItem> substrateConfigBuildItems,
 
             if (graalVmHome != null) {
                 Path graalVmLibDirectory = Paths.get(graalVmHome, "jre", "lib");
-                Path linuxLibDirectory = graalVmLibDirectory.resolve("amd64");
+                Path amd64LibDirectory = graalVmLibDirectory.resolve("amd64");
 
                 // We add . as it might be useful in a containerized world
                 // FIXME: it seems GraalVM does not support having multiple paths in java.library.path
                 //javaLibraryPathAdditionalPath.produce(new JavaLibraryPathAdditionalPathBuildItem("."));
-                if (Files.exists(linuxLibDirectory)) {
+                if (existsAndDoesNotContainOnlyTheAgent(amd64LibDirectory)) {
```

Comment:
Why don't we test the existence of the files? We could do something like:
- have a list of **file** candidates
- if exists, add the directory of the file to the library path

Reply:
The issue is that creates a tight dependency on a specific SSL related file, which is not great either.

Reply:
What's the exact file name on Linux?

Reply:
`/opt/graalvm/jre/lib/amd64/libsunec.so`

Maybe we could visit the `jre/lib/` hierarchy and try to find a file containing `sunec` (or starting with `libsunec`, I don't know the macOS name) in its name and then select the directory?

Reply:
Wondering about windows, so right now I'm working on finding the Linux one.

Reply:
@gsmet Can you check https://github.com/quarkusio/quarkus/pull/1681.